lib/Elevate/Components/NICs.pm

```diff
@@ -0,0 +1,98 @@
+package Elevate::Components::NICs;
+
+=encoding utf-8
+
+=head1 NAME
+
+Elevate::Components::NICs
+
+Rename NICs in the kernel namespace from ethX to cpethX
+
+=cut
+
+use cPstrict;
+
+use File::Slurper ();
+
+use Elevate::NICs ();
+
+use Log::Log4perl qw(:easy);
+
+use parent qw{Elevate::Components::Base};
+
+use constant ETH_FILE_PREFIX           => q[/etc/sysconfig/network-scripts/ifcfg-];
+use constant NIC_PREFIX                => q[cp];
+use constant PERSISTENT_NET_RULES_PATH => q[/etc/udev/rules.d/70-persistent-net.rules];
+
+sub pre_leapp ($self) {
+
+    $self->_rename_nics();
+
+    return;
+}
+
+sub _rename_nics ($self) {
+
+    # Only do this if there are multiple NICs in the kernel (eth) namespace
+    my @nics = Elevate::NICs::get_nics();
+    return unless scalar @nics > 1;
+
+    foreach my $nic (@nics) {
+        my $nic_path = ETH_FILE_PREFIX . $nic;
+
+        my $die_msg = <<~"EOS";
+        The file for the network interface card (NIC) using kernel-name ($nic) does
+        not exist at the expected location ($nic_path).  We are unable to
+        change the name of this NIC due to this.  You will need to resolve this
+        issue manually before elevate can continue.  Once the issue has been
+        resolved, you can continue this script by executing:
+
+        /scripts/elevate-cpanel --continue
+        EOS
+        die "$die_msg\n" unless -s $nic_path;
```

Comment:
I left this check in place because there is a possible race condition between the blocker and the component but I have also added a blocker for this scenario as well.